### PR TITLE
Set shell verbose output and group log messages by jobs

### PR DESCRIPTION
**Pull request by Blackhex**
Makes CI log slightly better readable by using groups, e.g. https://github.com/ZacWalk/mingw-woarm64-build/actions/runs/5576924316/jobs/10188914065. Enables shell commands printing.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#! /bin/bash
+set -e
+trap 'previous_command=$this_command; this_command=$BASH_COMMAND' DEBUG
```
Comment (Blackhex) [inline install.sh:3]: I've just copied the header and the footer from build.sh, though I don't fully understand what it's supposted to do.

Comment (ZacWalk) [reply install.sh:3]: The main goal was to exit the script if a command fails. And print out the command that failed.

Comment (ZacWalk) [reply install.sh:3]: No sure if the trap DEBUG is needed though.

Comment (Blackhex) [reply install.sh:3]: If I understand https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#index-trap-125 correctly, the \`DEBUG\` means is that the above will apply also for control statement commands.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -33,8 +33,8 @@ jobs:
         
       - name: Install Dependencies
         run: |
-          sudo apt update
-          sudo apt install -y build-essential binutils-for-build texinfo bison flex zlib1g-dev libgmp-dev dejagnu libmpc-dev
+          chmod +x ./install.sh
```
Comment (eukarpov) [inline .github/workflows/main.yml:36]: Could be avoided by adding permissions and adding changes to the repo on Linux

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -33,8 +33,8 @@ jobs:
...
+          bash ./install.sh
```
Comment (eukarpov) [inline .github/workflows/main.yml:37]: . instead of bash is an option

File: install.sh (hunk 1 above)
Comment (eukarpov) [reply install.sh:3]: Not sure it is needed here tracing is on and it should be enough

Comment (Blackhex) [reply install.sh]: JFYI, the install refactor was moved to a separate PR https://github.com/ZacWalk/mingw-woarm64-build/pull/34

**Comment by Blackhex** [comment]
This PR was quite outdated, and significant changes had to be done to make it current.

**Pull request merged**